modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/cell_renderers/DateTimeRenderer.tsx

```diff
@@ -21,15 +21,57 @@ function DateTimeRenderer({
 		return null;
 	}
 
+	let timestamp: number;
+
+	if (typeof value === 'string') {
+		const isoDateRegex =
+			/^(\d{4})-(\d{2})-(\d{2})(?:T(\d{2}):(\d{2}):(\d{2})Z)?$/;
+
+		const match = value.match(isoDateRegex);
+
+		if (match) {
+			const year = Number(match[1]);
+			const month = Number(match[2]) - 1;
+			const day = Number(match[3]);
+			const hours = Number(match[4] || 0);
+			const minutes = Number(match[5] || 0);
+			const seconds = Number(match[6] || 0);
+
+			timestamp = Date.UTC(year, month, day, hours, minutes, seconds);
+		}
+		else {
+			const date = value.split('T')[0];
+
+			const dateArray = date.split('-');
+
+			if (dateArray.length === 3) {
+				const [year, month, day] = dateArray;
+
+				timestamp = Date.UTC(
+					Number(year),
+					Number(month) - 1,
+					Number(day)
+				);
+			}
+			else {
+				timestamp = Number(value);
+			}
+		}
```

Comment:
Is this `else` necessary? Is assumes `YYYY-MM-DD` input, which we should not have any special handling in a `DateTimeRender`. That logic is for `DateRenderer`. Here, it may make sense to assure that the component does not break in this case, or in case of some other unexpected. format Maybe we can use here the unreliable Date parser as a fallback. Something like:
```js
if (match) {
    // controlled UTC parsing
}
else {
    timestamp = new Date(value);
}
```

Reply:
Makes sense, I made the changes.